transport.go

```diff
@@ -4,12 +4,42 @@ import (
 	"context"
 	"fmt"
 
+	addrutil "github.com/libp2p/go-addr-util"
 	host "github.com/libp2p/go-libp2p-host"
 	swarm "github.com/libp2p/go-libp2p-swarm"
 	tpt "github.com/libp2p/go-libp2p-transport"
 	ma "github.com/multiformats/go-multiaddr"
 )
 
+const P_CIRCUIT = 290
```

Comment:
I believe this constant needs to be added to the multicodec repo: https://github.com/multiformats/multicodec/